File: ruby/algorithms/insertion_sort.rb
def insertion_sort(array)
  (array.length).times do |j|
    while j > 0
      if array[j-1] > array[j]
        array[j], array[j-1] = array[j-1], array[j]
      else
        break
      end
      j-=1
    end
  end
  array
end

File: nim/algorithms/insertion_sort.nim
proc insertion_sort(arr: var seq[int]) = 
  for i in 0..<(arr.len - 1):
    var min = arr[i]
    var min_idx = i

    for j in i..<(arr.len):
      if arr[j] < min:
        min = arr[j]
        min_idx = j
    
    echo arr
    
    if min_idx != i:
      var temp = arr[i]
      arr[i] = arr[min_idx]
      arr[min_idx] = temp
  
var arr = @[5, 7, 4, 8, 2, 6, 3]

insertion_sort(arr)

echo arr